src/consumer.cpp

```diff
@@ -261,8 +261,8 @@ MessageList Consumer::poll_batch(size_t max_batch_size) {
 
 MessageList Consumer::poll_batch(size_t max_batch_size, milliseconds timeout) {
     vector<rd_kafka_message_t*> raw_messages(max_batch_size);
-    rd_kafka_queue_t* queue = rd_kafka_queue_get_consumer(get_handle());
-    ssize_t result = rd_kafka_consume_batch_queue(queue, timeout.count(), raw_messages.data(),
+    Queue queue(get_queue(rd_kafka_queue_get_consumer(get_handle())));
```

Comment:
This won't actually fix the issue if someone's using rdkafka version < 0.11.5 (see the comment on `get_queue`). Maybe it's worth adding a comment saying that right above this line? e.g. "Note that this will leak the queue when using rdkafka < 0.11.5 (see get_queue comment)`.

Reply:
Squashed the comment to the same commit: 
https://github.com/mfontanini/cppkafka/pull/107